runtime/finalise.c

```diff
@@ -145,6 +145,7 @@ void caml_final_do_calls (void)
   struct caml_final_info *fi = Caml_state->final_info;
 
   if (fi->running_finalisation_function) return;
+  Call_timing_hook(caml_finalise_begin_hook);
   if (fi->todo_head != NULL) {
```

Comment:
I think the hook should be called from within the test, not outside it.

Reply:
Well-spotted, thanks